docs/design/datacontracts/SyncBlock.md

```diff
@@ -0,0 +1,165 @@
+# Contract Object
+
+This contract is for getting information about SyncBlocks.
+
+## APIs of contract
+
+```csharp
+public readonly struct SyncBlockData
```

Comment:
I think we should hold off on this for a bit. @jkoritzinsky is reworking the monitors and how they exist on the sync block. See https://github.com/dotnet/runtime/pull/119335.